train.py

```diff
@@ -244,6 +244,15 @@ def loss_fn(pred, labels):
         states={"train_state": train_state},
         job_config=job_config,
     )
+
+    if job_config.checkpoint.create_seed_checkpoint:
+        assert (
+            world_size == 1
```

Comment:
I have some questions about the meta init workflow, assuming we are planning to train 70b model with PP. But initially we want to create a seed checkpoint so that we can use it later for meta init load. Do it mean that we need to init the 70b model on CPU first? If we init a debug model and save its seed checkpoint, I guess that won't be reusable in the later 70b model?

Reply:
the seed checkpoint must match the model we're training.

I suppose i have not even tested the seed creation on cpu vs gpu- probably we need to expose more options or smarts to determine which device to use here.

> Do it mean that we need to init the 70b model on CPU first?

yes, that's right. if this isn't OK, i think we can try to hand-write some initializer functions that can work with the PP traced function, but maybe that can come later?

Note: even if we say "ditch pipeline tracer", it wont fix the initializer problems.  We'd still need to customize the model's init_weights functions so that they can work given a 'model chunk' instead of a whole model.  And then we'd also have to tolerate some RNG divergence from the non-PP case, or, implement RNG seed passing and coordination.

Reply:
Yeah that can come later. My main worry is that for larger models like 70B/400B, this is going to be quite a initialization bottleneck, that we need to do:
1. create a 70B/400B model on CPU and save it to disk, probably take a couple of minutes
2. trace the 70B/400B model, take another tensor of couple of minutes
3. load the checkpoint, take another couple of minutes (this is fine I think as training anyways always want to save/load checkpoints)

Maybe we should brainstorming more on how PP could support meta init in a more sound way, i.e. maybe a PipelineStage.init_weights that wait for its RNG (if not the first stage), init the current meta model's weights, and transmit the RNGs to next stage

Reply:
> 2. trace the 70B/400B model, take another tensor of couple of minutes

do you think the tracing time will be long?  I actually assumed that the tracing time would not be an issue, it should be more related to the number of operators than the parameter size.

> Maybe we should brainstorming more on how PP could support meta init in a more sound way,
I am kind of ambivalent about this. Yes, i like the idea, even proposed it at one point.  But on the other hand I see quite a few more important issues to solve first.  And this idea is quite complicated, so it should be justified. So this seems like something P1 to me.